File: app/src/main/java/com/fakhrulasa/accountathenticator/authjava/AuthenticationService.java
package com.fakhrulasa.accountathenticator.authjava;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;

import com.fakhrulasa.accountathenticator.authentication.AccountAuthenticator;

/**
 * Class Name: AuthenticationService
 * Description: The AuthenticationService is the Authenticator Service for the application.
 */
public class AuthenticationService extends Service {
    public AuthenticationService() {
    }

    //region Overrides

    @Override
    public IBinder onBind(Intent intent) {

        AccountAuthenticator accountAuthenticator = new AccountAuthenticator(this);
        return accountAuthenticator.getIBinder();
    }

    //endregion
}

File: app/src/main/java/com/fakhrulasa/accountathenticator/authentication/AuthenticationService.kt
package com.fakhrulasa.accountathenticator.authentication

import android.app.Service
import android.content.Intent
import android.os.IBinder
import com.fakhrulasa.accountathenticator.authentication.AccountAuthenticator

/**
 * Class Name: AuthenticationService
 * Description: The AuthenticationService is the Authenticator Service for the application.
 */
class AuthenticationService : Service() {
    //region Overrides
    override fun onBind(intent: Intent): IBinder? {
        val accountAuthenticator = AccountAuthenticator(this)
        return accountAuthenticator.iBinder
    } //endregion
}